browser/brave_stats/brave_stats_updater_browsertest.cc

```diff
@@ -76,187 +85,117 @@ class BraveStatsUpdaterBrowserTest : public InProcessBrowserTest {
     env->SetVar("BRAVE_REFERRALS_LOCAL", "1");  // use http for local testing
   }
 
-  PrefService* GetLocalState() { return &testing_local_state_; }
-
   std::string GetUpdateURL() const { return update_url_; }
 
-  int WritePromoCodeFile(const std::string& referral_code) {
-    base::ScopedAllowBlockingForTesting allow_blocking;
-
-    base::FilePath user_data_dir;
-    base::PathService::Get(chrome::DIR_USER_DATA, &user_data_dir);
-
-    const base::FilePath promo_code_file =
-        user_data_dir.AppendASCII("promoCode");
-    return base::WriteFile(promo_code_file, referral_code.c_str(),
-                           referral_code.size());
-  }
+  void OnReferralInitialized(const std::string& referral_code) {
+    if (wait_for_referral_initialized_loop_) {
+      wait_for_referral_initialized_loop_->Quit();
+    }
 
-  void OnReferralInitialized(const std::string& download_id) {
-    referral_was_initialized_ = true;
-    // TODO(keur): Maybe do something with the download id
-    wait_for_callback_loop_->Quit();
+    on_referral_initialized_ = true;
+    referral_code_ = referral_code;
   }
 
   void WaitForReferralInitializeCallback() {
-    if (referral_was_initialized_)
+    if (wait_for_referral_initialized_loop_ || on_referral_initialized_) {
       return;
-    wait_for_callback_loop_.reset(new base::RunLoop);
-    wait_for_callback_loop_->Run();
+    }
+
+    wait_for_referral_initialized_loop_.reset(new base::RunLoop);
```

Comment:
Nit: better use `make_unique<T>()` instead of `reset + new` (here and in other cases below in this file)